hw11_cache/src/main/java/ru/otus/core/service/DbServiceUserCache.java

```diff
@@ -0,0 +1,30 @@
+package ru.otus.core.service;
+
+import ru.otus.cachehw.HwCache;
+import ru.otus.cachehw.MyCache;
+import ru.otus.core.dao.UserDao;
+import ru.otus.core.model.User;
+
+import java.util.Optional;
+
+public class DbServiceUserCache extends DbServiceUserImpl{
+
+    public HwCache<Long, User> cache = new MyCache();
+    public DbServiceUserCache(UserDao userDao) {
```

Comment:
```suggestion
    public DbServiceUserCache(UserDao userDao, HwCache<Long, User> cache) {
```

Реализацию кеша тоже лучше передавать как внешнюю зависимость